src/components/va-text-input/va-text-input.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  Component,
+  Element,
+  Prop,
+  Host,
+  h,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+
+@Component({
+  tag: 'va-text-input',
+  styleUrl: 'va-text-input.css',
+  shadow: true,
+})
+export class VaTextInput {
+  @Element() el: HTMLElement;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  /**
+   * The label for the text input.
+   */
+  @Prop() label: string | HTMLElement;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string | HTMLElement;
+
+  /**
+   * Set the input to required and render the (Required) text
+   */
+  @Prop() required?: boolean;
+
+  /**
+   * Placeholder text to show in the input field
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * The maximum number of characters allowed in the input.
+   */
+  @Prop() maxlength?: number;
+
+  /**
+   * What to tell the browser to auto-complete the field with.
+   */
+  @Prop() autocomplete?: string;
+
+  /**
+   * Don't emit any component-library-analytics events
+   */
+  @Prop() disableAnalytics?: boolean;
+
+  /*
+   * The value for the input.
+   */
+  @Prop({ mutable: true }) value?: string = '';
+  // TODO: Make the value prop reflective. Currently, it isn't because it screws
+  // up the input behavior. For now, the only bug is that the changed value
+  // isn't reflected in the DOM on the web component.
```

Comment:
Should we link to the MDN docs where it says that an `<input` value isn't _meant_ to be reflective?